src/mgmt/srv_drpc.c

```diff
@@ -355,6 +355,55 @@ create_pool_props(daos_prop_t **out_prop, char *owner, char *owner_grp,
 	return rc;
 }
 
+static int pool_create_fill_resp(Mgmt__PoolCreateResp *resp, uuid_t uuid, d_rank_list_t *svc_ranks)
+{
+	int			rc = 0;
+	d_rank_list_t	       *enabled_ranks = NULL;
+	daos_pool_info_t	pool_info = { .pi_bits = DPI_ENGINES_ENABLED | DPI_SPACE };
+
+	D_ASSERT(svc_ranks != NULL);
+	D_ASSERT(svc_ranks->rl_nr > 0);
+
+	rc = rank_list_to_uint32_array(svc_ranks, &resp->svc_reps, &resp->n_svc_reps);
+	if (rc != 0) {
+		D_ERROR("Failed to convert svc rank list: rc=%d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	D_DEBUG(DB_MGMT, "%d service replicas\n", svc_ranks->rl_nr);
+
+	rc = ds_mgmt_pool_query(uuid, svc_ranks, &enabled_ranks, &pool_info);
+	if (rc != 0) {
+		D_ERROR("Failed to query created pool: rc=%d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	rc = rank_list_to_uint32_array(enabled_ranks, &resp->tgt_ranks, &resp->n_tgt_ranks);
+	if (rc != 0) {
+		D_ERROR("Failed to convert enabled target rank list: rc=%d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	D_ASSERT(pool_info.pi_space.ps_space.s_total[DAOS_MEDIA_SCM] % resp->n_tgt_ranks == 0);
+	D_ASSERT(pool_info.pi_space.ps_space.s_total[DAOS_MEDIA_NVME] % resp->n_tgt_ranks == 0);
+	D_ALLOC_ARRAY(resp->tier_bytes, DAOS_MEDIA_MAX);
+	if (resp->tier_bytes == NULL) {
+		rc = -DER_NOMEM;
+		D_ERROR("Failed to allocate memory for tiers size: rc=%d\n", rc);
+		D_GOTO(out, rc);
+	}
+	resp->n_tier_bytes = DAOS_MEDIA_MAX;
+	resp->tier_bytes[DAOS_MEDIA_SCM] =
+		pool_info.pi_space.ps_space.s_total[DAOS_MEDIA_SCM] / resp->n_tgt_ranks;
+	resp->tier_bytes[DAOS_MEDIA_NVME] =
+		pool_info.pi_space.ps_space.s_total[DAOS_MEDIA_NVME] / resp->n_tgt_ranks;
```

Comment:
Seems like it would be better to do this in a loop, e.g. `for (i = 0; i < DAOS_MEDIA_MAX; i++)` in order to be future-proof for the addition of new tier types.

Reply:
I hesitated between a loop or doing it manually, and I finally preferred the second solution as it was simple: KISS principle ;-) However, I agree that doing it with the loop as you suggested is more future-proof.

- [ ] do tier_bytes assignment within a loop

Reply:
I don't see a downside to doing it in a loop. We already have a lot of technical debt due to old assumptions about only having two storage tiers, but we should be writing new code with the understanding that we are planning to support N storage tiers.